chart/k8gb/values.yaml

```diff
@@ -14,7 +14,8 @@ k8gb:
   dnsZone: "cloud.example.com" # dnsZone controlled by gslb
   dnsZoneNegTTL: 300 # Negative TTL for SOA record
   edgeDNSZone: "example.com" # main zone which would contain gslb zone to delegate
-  edgeDNSServer: "1.1.1.1" # use this DNS server as a main resolver to enable cross k8gb DNS based communication
+  edgeDNSServers: 
+      - "1.1.1.1" # use this DNS server as a main resolver to enable cross k8gb DNS based communication
```

Comment:
How does it behave with the old value? What customers will see when they realize we broke backward compatibility?

Reply:
good point, it's a breaking change so it's going to fail during the operator startup during the validation with:

```
2021-09-13T12:22:35Z ERR workspace/main.go:66 > can't resolve environment variables error="'EDGE_DNS_SERVERS' is empty"
```

So I might add some logic to `depresolver_config.go` that would produce a better message in case the `EDGE_DNS_SERVER` is set but `EDGE_DNS_SERVERS` is not.  We may also accept the old format and internally transform it to a list with just one entry and produce some warning htat this has been deprecated and use the new format because it may be removed in the new version. On the other hand it's still under `1.0.0` so breaking is allowed according to sem-ver :D

Another thing to consider is the helm chart discrepancy. Meaning when deploying newer version of k8gb w/ older helm chart by explicitly specifying the `k8gb.imageTag=somethingNewer` or vice-versa (updated chart and deploying some old image that had different "env. variable contract"). Handling it in the helm chart with conditions would make it quite unreadable and pain to maintain, imho

Reply:
depresolver to fail fast is a must imho. We can try internal conversion, that might be classy and actually very user friendly. Helm chart version might be irrelevant as the custom values are orthogonal to it and usually kept somewhere in gitops-like repo not directly connected to specific chart version.

Reply:
did that in https://github.com/k8gb-io/k8gb/pull/605/commits/f19dcaca3b857b22b5a2097bb28ea659a7646712

then it looks like this in the logs:

```
...
2021-09-13T14:45:17Z INF workspace/controllers/logging/logr.go:54 > metrics server is starting to listen {}
2021-09-13T14:45:17Z WRN workspace/main.go:88 > 'EDGE_DNS_SERVER' has been deprecated, use EDGE_DNS_SERVERS instead. Details: Pass the hostname or IP address as comma-separated list
2021-09-13T14:45:17Z INF workspace/main.go:91 > Registering components
...
```

So it doesn't fail, but transforms the old format into a new one + the warning in the logs in case of using the deprecated format, so we are now backward-compatible